app/codec/ffmpeg/ffmpegdecoder.cpp

```diff
@@ -350,6 +350,15 @@ FootageDescription FFmpegDecoder::Probe(const QString &filename, CancelAtom *can
 
     int64_t footage_duration = fmt_ctx->duration;
 
+    bool bad_duration = false;
+
+    // Catch when ffmpeg uses its inaccurate method of duration estimation so we can 
+    // use manual calculation
+    if (fmt_ctx->duration_estimation_method == AVFMT_DURATION_FROM_BITRATE) {
```

Comment:
This might be a little pedantic, but I would prefer a more descriptive variable name than `bad_duration`. Maybe something like `duration_guessed_from_bitrate`. You could also make this tidier by simply setting it from the initialization (i.e. `bool duration_guessed_from_bitrate = (fmt_ctx->duration_estimation_method == AVFMT_DURATION_FROM_BITRATE);`)

Reply:
Thanks for the feedback, much appreciated. I've updated as you suggested